src/android/Capture.java

```diff
@@ -399,32 +432,18 @@ public void onImageActivityResult(Request req) {
     }
 
     public void onVideoActivityResult(Request req, Intent intent) {
-        Uri data = null;
-
-        if (intent != null){
-            // Get the uri of the video clip
-            data = intent.getData();
-        }
-
-        if( data == null){
-            File movie = new File(getTempDirectoryPath(), "Capture.avi");
-            data = Uri.fromFile(movie);
-        }
-
-        // create a file object from the uri
-        if(data == null) {
+        if(this.videoAbsolutePath != null) {
+            req.results.put(createMediaFileWithAbsolutePath(this.videoAbsolutePath));
+        } else {
             pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: data is null"));
         }
-        else {
-            req.results.put(createMediaFile(data));
 
-            if (req.results.length() >= req.limit) {
-                // Send Uri back to JavaScript for viewing video
-                pendingRequests.resolveWithSuccess(req);
-            } else {
-                // still need to capture more video clips
-                captureVideo(req);
-            }
+        if (req.results.length() >= req.limit) {
+            // Send Uri back to JavaScript for viewing video
+            pendingRequests.resolveWithSuccess(req);
+        } else {
+            // still need to capture more videos
+            captureVideo(req);
         }
     }
 
```

Comment:
Should `createMediaFile(Uri data)` be removed?  It doesn't appear to be used / replaced with `createMediaFileWithAbsolutePath(String path)`

Reply:
@chriskhongqarma Sorry if this is the incorrect place for this comment.

Reply:
+1 as `createMediaFile(Uri data)` is not used anymore here, fully replaced by `createMediaFileWithAbsolutePath(String path)`

Reply:
and other unused elements now like `queryImgDB`, `checkForDuplicateImage`, `whichContentStore`, `numPics` that should be removed too